### PR TITLE
🎨 Palette: Ritual Progress Counter & Tactile Feedback

### PR DESCRIPTION
**Pull request by hu3mann**
This PR implements "Ritual Progress Awareness" in the TaskSequencer component, a micro-UX enhancement designed to ground ADHD users in their current workflow sequence.

### 💡 What:
- Introduced a `completedCount/totalCount` indicator in the TaskSequencer header and footer.
- Enhanced `ListItem` interactivity with a subtle vertical lift (`-2px`), deeper shadow, and border color shift on `:hover` or `:focus-within`.
- Added a `done-glow` keyframe animation to the "DONE" state status indicator.
- Updated `aria-label` attributes to explicitly announce task progress (e.g., "1/3 tasks completed").

### 🎯 Why:
- Users with ADHD often experience "spatial/temporal confusion" when navigating sequential tasks. Explicit progress fractions provide a concrete anchor for where they are in the ritual.
- Tactile visual feedback on list items makes the interface feel more responsive and reduces the cognitive gap between viewing and interacting.
- A distinctive "completion" state (DONE with glow) provides satisfying positive reinforcement.

### ♿ Accessibility:
- Updated ARIA labels for `role="status"` containers to include task counts.
- Ensured all interactive lift effects are applied on `:focus-within` to support keyboard users.
- Updated Vitest suite in `ui-dashboard/src/components/__tests__/Accessibility.test.ts` to ensure assertions remain green.

---
*PR created automatically by Jules for task [18337158806223683382](https://jules.google.com/task/18337158806223683382) started by @hu3mann*

### CODE REVIEW

**Review by hu3mann — COMMENTED** [review]
## Review (Claude Opus 4.7)

### Summary
Adds a `n/m tasks` progress counter and tactile lift/shadow hover feedback to `TaskSequencer.tsx`, plus a palette retro entry documenting the rationale. Follow-up commit `cf8d1cf80` on this branch addresses the Copilot review concerns by routing the six "done" visual cues through a real-completion boolean and fixing the hover shadow clipping.

### Strengths
- **Correct rationale.** The palette entry frames this as ADHD spatial/temporal grounding and tactile interactivity; that's the right framing for the change.
- **Memoised counts.** `completedCount`/`totalCount` are derived in a `useMemo` keyed on `tasks`, no per-render churn.
- **Hover lift uses cubic-bezier transition** matching existing motion language in this file.

### Fixes pushed in cf8d1cf80
- **Real completion check.** Introduced `isComplete = totalCount > 0 && completedCount === totalCount` and routed all six "done" visual cues through it (title, ARIA role="status" label, color, done-glow animation, focus-visible boxShadow, icon swap, DONE text). Previously these all keyed off `displayRemainingMinutes === 0`, which fires when the current task overruns its estimate but later tasks remain — announcing "sequence complete" while work was queued. Both block instances (header timer + Optimized Sequence header) updated.
- **Hover shadow clipping.** Added `position: 'relative'` + `zIndex: 1` to the ListItem `&:hover, &:focus-within` rule so the `0 4px 12px` drop shadow lifts above the next sibling instead of being clipped by it.

### Observations / non-blocking
- **Counter format consistency.** "n/m • Xm" in the badge vs "n/m tasks • durationLabel" in the tooltip is fine; the slight asymmetry trades brevity for badge legibility. Worth a screenshot review before final approval.
- **`done-glow` keyframes are duplicated** between the two visual blocks. Functional (MUI dedupes at runtime) but a future cleanup could lift them to a shared constant.
- **Empty-task edge case.** With `totalCount === 0`, `isComplete` is correctly `false`, so the UI falls back to the time-based path. The `>0` guard avoids the "0/0 ⇒ DONE" trap.

### Tests
- No tests touch TaskSequencer behavior in this PR; the existing `Accessibility.test.ts` doesn't assert on completion text. A unit test asserting "displayRemainingMinutes === 0 with incomplete tasks does NOT announce 'Task sequence complete'" would lock the new behavior.

### Verdict
👍 Approve after CI re-runs against cf8d1cf80. Visual confirmation in dev (hover lift, completion announcement timing) recommended before merge.

File: ui-dashboard/src/components/TaskSequencer.tsx [hunk 1]
```diff
@@ -263,7 +270,7 @@ const TaskSequencer: React.FC<TaskSequencerProps> = ({ cognitiveState }) => {
           title={
             displayRemainingMinutes === 0
               ? 'Task sequence complete'
-              : `${getDurationAriaLabel(displayRemainingMinutes)} (${finishTimeLabel})`
+              : `${completedCount}/${totalCount} tasks • ${getDurationAriaLabel(displayRemainingMinutes)} (${finishTimeLabel})`
           }
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply ui-dashboard/src/components/TaskSequencer.tsx:277]: Addressed in cf8d1cf80: routed the completion title through `isComplete = totalCount > 0 && completedCount === totalCount` instead of `displayRemainingMinutes === 0`.

File: ui-dashboard/src/components/TaskSequencer.tsx [hunk 2]
```diff
@@ -272,7 +279,7 @@ const TaskSequencer: React.FC<TaskSequencerProps> = ({ cognitiveState }) => {
             aria-label={
               displayRemainingMinutes === 0
                 ? 'Task sequence complete'
-                : `${getDurationAriaLabel(displayRemainingMinutes)}. Estimated completion: ${finishTimeLabel}`
+                : `${completedCount}/${totalCount} tasks completed. ${getDurationAriaLabel(displayRemainingMinutes)}. Estimated completion: ${finishTimeLabel}`
             }
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply ui-dashboard/src/components/TaskSequencer.tsx:286]: Addressed in cf8d1cf80: the `role="status"` aria-label now keys on `isComplete`.

File: ui-dashboard/src/components/TaskSequencer.tsx [hunk 3]
```diff
@@ -282,7 +289,14 @@ const TaskSequencer: React.FC<TaskSequencerProps> = ({ cognitiveState }) => {
               gap: 0.5,
               color: displayRemainingMinutes === 0 ? brandTokens.colors.serumMint : brandTokens.colors.saintGold,
               cursor: 'help',
-              transition: 'color 0.3s ease',
+              transition: 'all 0.3s ease',
+              ...(displayRemainingMinutes === 0 && {
+                animation: 'done-glow 2s infinite ease-in-out',
+                '@keyframes done-glow': {
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply ui-dashboard/src/components/TaskSequencer.tsx:298]: Addressed in cf8d1cf80: the done-glow animation now triggers on `isComplete`.

File: ui-dashboard/src/components/TaskSequencer.tsx [hunk 4]
```diff
@@ -539,7 +553,7 @@ const TaskSequencer: React.FC<TaskSequencerProps> = ({ cognitiveState }) => {
           title={
             displayRemainingMinutes === 0
               ? 'Task sequence complete'
-              : `${getDurationAriaLabel(displayRemainingMinutes)} (${finishTimeLabel})`
+              : `${completedCount}/${totalCount} tasks • ${getDurationAriaLabel(displayRemainingMinutes)} (${finishTimeLabel})`
           }
           arrow
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply ui-dashboard/src/components/TaskSequencer.tsx]: Addressed in cf8d1cf80: the completion indicator now uses `isComplete`.

File: ui-dashboard/src/components/TaskSequencer.tsx [hunk 5]
```diff
@@ -551,7 +565,10 @@ const TaskSequencer: React.FC<TaskSequencerProps> = ({ cognitiveState }) => {
               ml: 'auto',
               color: displayRemainingMinutes === 0 ? brandTokens.colors.serumMint : brandTokens.colors.ritualCyan,
               cursor: 'help',
-              transition: 'color 0.3s ease',
+              transition: 'all 0.3s ease',
+              ...(displayRemainingMinutes === 0 && {
+                animation: 'done-glow 2s infinite ease-in-out',
+              }),
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply ui-dashboard/src/components/TaskSequencer.tsx]: Addressed in cf8d1cf80: the second done-glow gate (Optimized Sequence header) also uses `isComplete`. Both blocks use the same keyframes locally; functional today, candidate for a future hoist.

File: ui-dashboard/src/components/TaskSequencer.tsx [hunk 6]
```diff
@@ -614,11 +631,14 @@ const TaskSequencer: React.FC<TaskSequencerProps> = ({ cognitiveState }) => {
                     ? `1px solid ${brandTokens.borders.cyan}`
                     : `1px solid ${brandTokens.borders.subtle}`,
                   mb: 0.5,
-                  transition: 'background-color 0.2s ease',
-                  '&:hover': {
+                  transition: 'all 0.2s cubic-bezier(0.4, 0, 0.2, 1)',
+                  '&:hover, &:focus-within': {
                     bgcolor: isCurrent
                       ? alpha(brandTokens.colors.ritualCyan, 0.12)
                       : alpha(brandTokens.colors.ritualCyan, 0.04),
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply ui-dashboard/src/components/TaskSequencer.tsx:641]: Addressed in cf8d1cf80: added `position: 'relative'` and `zIndex: 1` on the ListItem `&:hover, &:focus-within` rule so the lift drop-shadow stacks above the next sibling.

File: ui-dashboard/src/components/__tests__/Accessibility.test.ts [hunk 7]
```diff
@@ -74,13 +74,13 @@ test('TaskSequencer.tsx has contextual aria-labels and current step indicator',
   expect(content).toContain('aria-label={getTimerAriaLabel(taskTimer)}');
   // Total remaining duration
   expect(content).toContain('role="status"');
-  expect(content).toMatch(/aria-label=\{\s*displayRemainingMinutes === 0\s*\?\s*'Task sequence complete'\s*:\s*`\$\{getDurationAriaLabel\(displayRemainingMinutes\)\}\. Estimated completion: \$\{finishTimeLabel\}`\s*\}/);
+  expect(content).toMatch(/aria-label=\{\s*displayRemainingMinutes === 0\s*\?\s*'Task sequence complete'\s*:\s*`\$\{completedCount\}\/\$\{totalCount\} tasks completed\. \$\{getDurationAriaLabel\(displayRemainingMinutes\)\}\. Estimated completion: \$\{finishTimeLabel\}`\s*\}/);
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply ui-dashboard/src/components/__tests__/Accessibility.test.ts]: Addressed in b796b859e: updated the regex matchers in Accessibility.test.ts to match the new `isComplete` branches and the ETA-guarded format. `vitest run src/components/__tests__/Accessibility.test.ts` passes 10/10.

[the remark this answers is not in the document]
Comment (hu3mann) [reply ui-dashboard/src/components/__tests__/Accessibility.test.ts]: Addressed in b796b859e: same regex update as the sibling thread; matchers now branch on `isComplete` rather than `displayRemainingMinutes === 0`.

File: ui-dashboard/src/components/TaskSequencer.tsx [hunk 8]
```diff
@@ -261,42 +274,49 @@ const TaskSequencer: React.FC<TaskSequencerProps> = ({ cognitiveState }) => {
         </Typography>
         <Tooltip
           title={
-            displayRemainingMinutes === 0
+            isComplete
               ? 'Task sequence complete'
-              : `${getDurationAriaLabel(displayRemainingMinutes)} (${finishTimeLabel})`
+              : `${completedCount}/${totalCount} tasks • ${getDurationAriaLabel(displayRemainingMinutes)} (${finishTimeLabel})`
```
[the remark this answers is not in the document]
Comment (hu3mann) [reply ui-dashboard/src/components/TaskSequencer.tsx:276]: Addressed in b796b859e: guarded both tooltip title and aria-label so the `(${finishTimeLabel})` / `Estimated completion: ${finishTimeLabel}` fragments only render when `finishTimeLabel` is non-empty.

**Comment by hu3mann** [comment]
Closing this PR — the head branch has been corrupted by an automated commit (`14ebfb09d` from `google-labs-jules[bot]`) that re-introduces work already deleted from `main` and deletes work freshly merged into `main`.

If squash-merged in current state, this PR would touch **224 files**:
- **Re-add ~68 files** that were intentionally removed in #586 (`services/genetic_agent/**`, `docker/mcp-servers-source/task-master-ai/**`, `scripts/jules_submit_pm_plane_packets.py`, `*.bak` files, etc.)
- **Delete ~62 files** that other PRs added: `out/cockpit-design-pickup/**` (#597), `out/cockpit-main-state-recon/**`, `out/cockpit-merge-execute/**`, `.github/security-filtering-instructions.txt` and `security-scan-instructions.txt` (#588).
- **Revert ~94 modifications** the recent batch of merges produced: workflow path fixes, `compose.yml`, `desktop-commander/server.py` (FastMCP v3 migration), `CHANGELOG.md`, `INSTALL.md`, etc.

The legitimate ritual-progress work (`.Jules/palette.md`, `ui-dashboard/src/components/TaskSequencer.tsx`, `ui-dashboard/src/components/__tests__/Accessibility.test.ts`) can be re-filed cleanly on a fresh branch from current `main` — including:
- The `n/m tasks` progress counter and tactile lift/shadow hover feedback (original PR scope),
- The `isComplete = totalCount > 0 && completedCount === totalCount` refactor that routes the six "done" visual cues through real completion state instead of `displayRemainingMinutes === 0`,
- The ETA guard for empty `finishTimeLabel`,
- The `position: 'relative'` + `zIndex: 1` lift-shadow stacking fix,
- The matching `Accessibility.test.ts` regex updates.
